### PR TITLE
implement a `PageRegion` structure

### PR DESCRIPTION
**Pull request by rphmeier**
This implements a `PageRegion` that fulfills the requirements of the `rw_pass_cell::Region` trait.

### CODE REVIEW

File: nomt/src/page_region.rs [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+use crate::rw_pass_cell::{Region, RegionContains};
+
+/// A region of pages for shared/exclusive access.
+///
+/// See module docs for more details.
+#[derive(Debug, Clone)]
+pub struct PageRegion {
+    // non-exclusive access: if min is `None` any page along the path to this one (excluding this).
+    //                       otherwise, including this page.
+    // exclusive access: if min is `None`, just to this page ID. otherwise none.
```
Comment (gabriele-0201) [inline nomt/src/page_region.rs:52]: shouldn't it be: `exclusive access: if min is None, from this page ID. otherwise from min`

**Review by gabriele-0201 — APPROVED** [review]

**Comment by gabriele-0201** [comment]
The exclusive_min option is really nice for indicating whether the pointed page has exclusive access or not! Not talking about PageId ordering, it makes the implementation super clean!

File: nomt/src/page_region.rs (hunk 1 above)
Comment (rphmeier) [reply nomt/src/page_region.rs:52]: ah, this comment was meant to just be on this variable, but I will edit for clarity

**Review by pepyakin — APPROVED** [review]
Ideally we settle on the same terminology re shared/non-exclusive & exclusive and read & write/mutable.

**Pull request merged**